### PR TITLE
Added support for latest pytorch-lightning, albumentation and fixed multiple issues with DDP training, images sizes normalization and focal loss

### PR DESCRIPTION
**Pull request by ExtReMLapin**
* Added support for pytorch lightning and it's latest version,
    * updated few methods names and added options like `ddp_find_unused_parameters_false`
* Fixed crash with batch size != 1 with images of different sizes using padding (on square image) and `--max_image_size` which defaults to 512
    * i could have added smarter padding instead of box of size 512x512
* implemented focal loss using `--use_focal_loss`
* fixed multiple errors with ddp
   * crashes and errors linked to tensors being split on multiple devices including CPU
   * slow start up because of non recycled dataloaders (recreated on each epoch)
   * images not uploaded to wandb because of missing ENV variable
* implemented better schedulers

Please note that when increasing image size you obviously need to increase points size (`--heatmap_sigma`) and other evaluation metrics like 
* `--minimal_keypoint_extraction_pixel_distance`
* `--maximal_gt_keypoint_pixel_distances`

### CODE REVIEW

**Comment by ExtReMLapin** [comment]
Pr is still draft because things need to be cleaned up a bit like the gitignore and my very aggresive albumentations options
